Apps/Common/src/Services/AccessTokenService.cs

```diff
@@ -0,0 +1,172 @@
+            TokenSwapResponse? cachedAccessToken = this.GetFromCache(hdid);
+
+            if (cachedAccessToken == null)
+            {
+                string? accessToken = this.authenticationDelegate.FetchAuthenticatedUserToken();
+
+                if (accessToken != null)
+                {
+                    requestResult = await this.SwapToken(hdid, accessToken).ConfigureAwait(true);
+                }
+                else
+                {
+                    this.logger.LogError("Unable to get authenticated user token from context for hdid: {Hdid}", hdid);
+                    requestResult.ResultStatus = ResultType.Error;
+                    requestResult.ResultError = new()
+                    {
+                        ResultMessage = $"Internal Error: Unable to get authenticated user token from context for hdid: '{hdid}'",
+                        ErrorCode = ErrorTranslator.InternalError(ErrorType.InvalidState),
+                    };
+                }
+            }
+            else
+            {
+                this.logger.LogDebug("Access token fetched from Cache");
+                requestResult.ResourcePayload = cachedAccessToken;
+                requestResult.ResultStatus = ResultType.Success;
+            }
+
+            return requestResult;
+        }
+
+        /// <summary>
+        /// Caches the Access Token if enabled.
+        /// </summary>
+        /// <param name="hdid">The hdid associated with the token swap response to be cached.</param>
+        /// <param name="tokenSwapResponse">The token swap response to be cached.</param>
+        private void CacheAccessToken(string hdid, TokenSwapResponse tokenSwapResponse)
+        {
+            using Activity? activity = Source.StartActivity();
+            string cacheKey = $"{TokenSwapCacheDomain}:HDID:{hdid}";
+            if (this.tokenCacheMinutes > 0)
+            {
+                this.logger.LogDebug("Attempting to cache access token for cache key: {Key}", cacheKey);
+                this.cacheProvider.AddItem(cacheKey, tokenSwapResponse, TimeSpan.FromMinutes(this.tokenCacheMinutes));
+            }
+            else
+            {
+                this.logger.LogDebug("Access token caching is disabled; will not cache for cache key: {Key}", cacheKey);
+            }
+
+            activity?.Stop();
+        }
+
+        /// <summary>
+        /// Attempts to get the token response from cache.
+        /// </summary>
+        /// <param name="hdid">The hdid used to create the key to retrieve the token response from cache.</param>
+        /// <returns>The found token response or null.</returns>
+        private TokenSwapResponse? GetFromCache(string hdid)
+        {
+            using Activity? activity = Source.StartActivity();
+            TokenSwapResponse? tokenResponse = null;
+            if (this.tokenCacheMinutes > 0)
+            {
+                string cacheKey = $"{hdid}-{TokenSwapCacheDomain}";
```

Comment:
Shouldn't this value match the one on line 120?

Reply:
Fixed. Forgot to change that one when I edited the cacheKey this morning.